### PR TITLE
Lazy load cohort data, load all on column click

### PR DESCRIPTION
**Pull request by adamabeshouse**
# Checks
- [x] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [x] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

### CODE REVIEW

**Comment by adamabeshouse** [comment]
[(5) Mutation Table --> Lazy load cohort column](https://trello.com/c/2igXSRNi/117-5-mutation-table-lazy-load-cohort-column)

**Comment by adamabeshouse** [comment]
[(8) Mutation Table --> when a user clicks on the cohort column, go get all the data.](https://trello.com/c/9AlHA266/118-8-mutation-table-when-a-user-clicks-on-the-cohort-column-go-get-all-the-data)

File: src/pages/patientView/clinicalInformation/MrnaExprRankCache.ts [hunk 1]
```diff
@@ -5,81 +5,78 @@ import internalClient from "../../../shared/api/cbioportalInternalClientInstance
...
-                cache[sampleId] = cache[sampleId] || {};
-                for (const entrezGeneId of entrezGeneIds) {
-                    if (!cache[sampleId].hasOwnProperty(entrezGeneId)) {
-                        toQuery[sampleId] = toQuery[sampleId] || [];
-                        toQuery[sampleId].push(entrezGeneId);
-                        cache[sampleId][entrezGeneId] = { status:"pending" };
-                    }
+        // See which we need to fetch, and set "pending" for those data
+        const toQuery:{ [sampleId:string]:number[]} = {};
+        _.forEach(sampleToEntrezGeneIds, (entrezGeneIds:number[], sampleId:string) => {
```
Comment (alisman) [inline src/pages/patientView/clinicalInformation/MrnaExprRankCache.ts]: sorry, i keep commenting on these iterators.  probably the array method forEach would stand the best chance of getting TS to infer the type.

File: src/pages/patientView/clinicalInformation/MrnaExprRankCache.ts [hunk 2]
```diff
@@ -5,81 +5,78 @@ import internalClient from "../../../shared/api/cbioportalInternalClientInstance
...
-                            if (!haveData[sampleId] || !haveData[sampleId][entrezGeneId]) {
-                                cache[sampleId][entrezGeneId] = { status: "not available" };
-                            }
+
+            const mrnaPercentiles: MrnaPercentile[]
+                = allMrnaPercentiles.reduce((arr:MrnaPercentile[], next:MrnaPercentile[])=>arr.concat(next));
+            const haveData:{ [sampleId:string]: { [entrezGeneId: string]:boolean}} = {};
+            for (const mrnaPercentile of mrnaPercentiles) {
+                // Add data
+                cache[mrnaPercentile.sampleId] || {};
```
Comment (alisman) [inline src/pages/patientView/clinicalInformation/MrnaExprRankCache.ts]: i think you need assign something to this, right?

**Pull request closed without merging**